libs/types/src/election_utils.ts

```diff
@@ -600,17 +601,18 @@ function preprocessElection(value: unknown): unknown {
 }
 
 /**
- * Parses `value` as an `Election` encoded as a JSON string. Equivalent to
- * `safeParseJson(Election, value)`.
+ * Parses `value` as a VXF `Election` object.
  */
-export function safeParseElection(
-  value: string
-): Result<Election, z.ZodError | SyntaxError>;
+export function safeParseVxfElection(
+  value: unknown
+): Result<Election, z.ZodError> {
+  return safeParse(ElectionSchema, preprocessElection(value));
+}
+
 /**
- * Parses `value` as an `Election` object. Equivalent to
- * `safeParse(Election, value)`.
+ * Parses `value` as an `Election` object. Supports both VXF and CDF. If given a
+ * string, will attempt to parse it as JSON first.
```

Comment:
So I'm pretty sure that after this change, not only does VxAdmin now support importing CDF, but the resulting `ElectionDefinition` will have the CDF raw JSON as its `electionData` (which will also be the data that is hashed). I think this means that the CDF will be saved in the db and propagated throughout the rest of the system (e.g. via smartcards/ballot packages). So maybe all of VxSuite now supports CDF ballot definitions? Am I missing something?

Reply:
- CDF will be saved to the DB

Reply:
My understanding is the same, after taking a look at the relevant code:

- CDF will be saved to the DB
- CDF will be saved to card. With the current cards there is the space concern but with @arsalansufi's new cards that is changing?
- CDF will be in ballot package
- CDF election hash will be encoded into the ballots

I don't know of anything we're missing!